documentation/source/newsfragments/2203.removal.rst

```diff
@@ -0,0 +1 @@
+Deprecate :func:`~cocotb.utils.pack` and :func:`~cocotb.utils.unpack` and the use of :class:`python:ctypes.Structure` in signal assignments.
```

Comment:
Can you add a test to demonstrate the deprecated signal assignment?

Reply:
Added. Will also deprecate the acceptance of `dict` values (which is totally undocumented) as a separate PR.